### PR TITLE
Feat/edit existing card

### PR DESCRIPTION
**Pull request by mmcode9407**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
Implement editing existing flashcards

### Requirements
1. It should be able to turn on `Edit` mode when clicking on a pencil icon
2. It should be able to `Cancel` editing
3. It should be able to `Save` editing

### Review

1. The code changes cover the requirement of turning on `Edit` mode when clicking on a pencil icon. This is implemented in the `FlashCardSide.tsx` file where an edit button is added and an `onClick` event is set to call the `handleEdit` function.

2. The code changes cover the requirement of being able to `Cancel` editing. This is implemented in the `FlashCardSide.tsx` file where a cancel button is added and an `onClick` event is set to call the `handleCancel` function.

3. The code changes cover the requirement of being able to `Save` editing. This is implemented in the `FlashCardSide.tsx` file where a save button is added and an `onClick` event is set to call the `handleSave` function.

4. The code changes also include some refactoring and cleaning up of the code, which makes the code more readable and maintainable.

5. The code changes include adding types for better type safety and autocompletion features.

6. The code changes include handling the case where there are no flashcards, by displaying a message "Add your first flashcard". This was not part of the requirements, but it's a good addition for better user experience.

In conclusion, the code changes cover all the task requirements and also include some additional improvements. There are no apparent bugs in the code changes.
